src/WebJobs.Extensions.DurableTask/DefaultPlatformInformation.cs

```diff
@@ -37,7 +38,7 @@ public DefaultPlatformInformation(INameResolver nameResolver, ILoggerFactory log
             if (!this.cachedEnviromentVariables.TryGetValue(variableName, out value))
             {
                 value = this.nameResolver.Resolve(variableName);
-                this.cachedEnviromentVariables.Add(variableName, value);
+                this.cachedEnviromentVariables.TryAdd(variableName, value);
```

Comment:
Should we add a log if the variable already exists (i.e. `TryAdd()` returns `false`)? Would that be helpful for investigations?

Reply:
I'm not sure this log would be very informative, let me explain why.

In this path, we're caching the values of environment variables (representing app settings) in a dictionary. This is simply so that, if we have read the variable in the past, we don't have to do another IO operation again.

Since we don't support dynamically changing app settings (at least not without a restart), if we already have a dictionary entry for a given app setting, then we can ignore subsequent attempts at caching that value because the entry we already have, and the one we would replace it with, are the same.

Finally, this codepath has always been concurrent, so it's possible that two threads might race to try to update this dictionary, but since they are both guaranteed to read the same value, I don't think it's too useful to log that a given thread lost the update race. I think it's safe to silently ignore because it's idempotent :) .

Reply:
That makes sense, thanks for the explanation!

Reply:
That makes sense, thanks for the explanation!